frontend/ui/widget/qrwidget.lua

```diff
@@ -0,0 +1,43 @@
+--[[
+QRWidget shows a QR code for a given text.
+]]
+
+local Blitbuffer = require("ffi/blitbuffer")
+local ImageWidget = require("ui/widget/imagewidget")
+local logger = require("logger")
+local qrencode = require("ffi/qrencode")
+local _ = require("gettext")
+
+local QRWidget = ImageWidget:extend{
+    scale_factor = 0,
+    text = ""
+    -- see ImageWidget for other options.
+}
+
+function QRWidget:init()
+    local text = self.text
+    if #text > 2953 then
+        local truncated = _('... (truncated...)')
+        text = text:sub(1, 2953 - #truncated) .. truncated
+    end
+    local ok, grid = qrencode.qrcode(text)
+    if not ok then
+        logger.info("QRWidget: failed to generate QR code.")
+        return
+    else
+        local scale = 5
```

Comment:
I'd rather this weren't hard-coded, but instead computed in a DPI-aware fashion via `Screen:scaleBySize`.

c.f., https://github.com/koreader/koreader/blob/master/frontend/ui/size.lua for examples ;).

Reply:
Just fixed ; thanks to your link, I discovered that there were default sizes, so I changed qrmessage to use default padding.